modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -130,52 +151,89 @@ inline int toCV(IE::Precision prec) {
-    // The reason for this is that RGB images are also considered 2D
-    // as the channel component is not taken into account in this case.
-    if (mat_desc.isND() || mat_desc.planar) {
-        return cv::gapi::ie::TraitAs::TENSOR;
-    }
-    // NB: If user provided 2D data in U8 precision
-    // and network expects NHWC/NCHW layout - data is image (most likely).
-    const auto layout = tensor_desc.getLayout();
-    if (layout == IE::Layout::NCHW || layout == IE::Layout::NHWC) {
-        if (mat_desc.depth == CV_8U) {
-            return cv::gapi::ie::TraitAs::IMAGE;
+inline IE::ResizeAlgorithm toIEInterp(int interpolation) {
+    switch (interpolation) {
+        case cv::INTER_LINEAR: return IE::RESIZE_BILINEAR;
+        case cv::INTER_AREA:   return IE::RESIZE_AREA;
+        default: GAPI_Error("IE Backend: Unsupported resize algorithm");
+    }
+    // Unreachable code
+    GAPI_Assert(false);
+}
+
+template <typename Attr>
+using AttrMap = cv::gapi::ie::detail::AttrMap<Attr>;
+
+template <typename Attr>
+using LayerVariantAttr = cv::gapi::ie::detail::LayerVariantAttr<Attr>;
+
+template <typename Attr> AttrMap<Attr>
+broadcastLayerAttr(const LayerVariantAttr<Attr>   &layer_attr,
+                   const std::vector<std::string> &layer_names) {
+    AttrMap<Attr> map;
+    if (cv::util::holds_alternative<AttrMap<Attr>>(layer_attr)) {
+        map = cv::util::get<AttrMap<Attr>>(layer_attr);
+        // NB: Validate map:
+        std::unordered_set<std::string> existing_layers =
+            {layer_names.begin(), layer_names.end()};
+
+        for (const auto &p : map) {
+            const auto it = existing_layers.find(p.first);
+            if (it == existing_layers.end()) {
+                cv::util::throw_error(
+                        std::logic_error("IE Backend: Failed to"
+                                         " find layer with name: " + p.first));
+            }
         }
-        // NB: 2D data with precision != U8 provided for 4D layout.
-        std::stringstream ss;
-        ss << "IE Backend: An inconsistency has been detected between"
-              " the provided data: " << mat_desc <<
-              " and the network layout: " << layout;
-        cv::util::throw_error(std::logic_error(ss.str()));
-    }
-    // NB: Otherwise trait is TENSOR.
-    // If there is an inconsistency between the data dimensions
-    // and the network layout, the "setBlob" will fail.
+    } else if (cv::util::holds_alternative<Attr>(layer_attr)) {
+        // NB: Broadcast value to all layers.
+        auto elem = cv::util::get<Attr>(layer_attr);
+        for (auto &&layer_name : layer_names) {
+            map.emplace(layer_name, elem);
+        }
+    }
+    return map;
+}
+
+template <typename K, typename V>
+cv::optional<V> lookUp(const std::map<K, V> &map, const K& key) {
+    const auto it = map.find(key);
+    if (it == map.end()) {
+        return {};
+    }
+    return cv::util::make_optional(std::move(it->second));
+}
```

Comment:
I'd add a TODO to move it to utils later. Because why not

Reply:
Added